### PR TITLE
Fixes #38117 - Refresh smart proxy sync history when proxy is updated

### PR DESCRIPTION
**Pull request by sjha4**
#### What are the changes introduced in this pull request?
Refresh smart proxy sync history when sp is updated and only keep sync histories of repos available to capsule.
#### Considerations taken when implementing this change?
When LCE is removed from sp, we will now assume the sync history is not needed anymore.
What this does is that when env is added back, an optimized sync will not skip over this repo during sync.
#### What are the testing steps for this pull request?
1. Sync a LCE to the Smart proxy. This shows the package counts for cv correctly.
2. Now remove the LCE from Smart proxy.
3. Now add the previous LCE to the Smart proxy and sync it.
4. Package count now shows up as N/A
5. Refresh count doesnt fix this issue. Only complete sync fixes the counts again.

The issue here is that optimized sync simply skips the repos from the environment cause their old sync histories have not been cleared.

### CODE REVIEW

File: app/models/katello/concerns/smart_proxy_extensions.rb [hunk 1]
```diff
@@ -479,6 +483,17 @@ def remove_lifecycle_environment(environment)
...
 
+      def refresh_smart_proxy_sync_histories
+        smart_proxy_helper = ::Katello::SmartProxyHelper.new(self)
+        repos = smart_proxy_helper.repositories_available_to_capsule
+        repo_ids = repos.map(&:id)
+        if repo_ids.empty?
+          self.smart_proxy_sync_histories.delete_all
+        else
+          self.smart_proxy_sync_histories.where("repository_id NOT IN (?)", repo_ids).delete_all
+        end
```
Comment (ianballou) [inline app/models/katello/concerns/smart_proxy_extensions.rb:494]:
```suggestion
        repos = smart_proxy_helper.repositories_available_to_capsule.select(:id)
        # Use size to ensure repos don't get loaded unnecessarily
        if repos.size == 0
          self.smart_proxy_sync_histories.delete_all
        else
          # Uses a single SQL query without needing to load IDs
          self.smart_proxy_sync_histories.where(id: repos).delete_all
        end
```

**Review by ianballou — COMMENTED** [review]
Working well! Just one comment about calculating the sync histories to delete.

File: app/models/katello/concerns/smart_proxy_extensions.rb (hunk 1 above)
Comment (sjha4) [reply app/models/katello/concerns/smart_proxy_extensions.rb]: Ack..Updated to this with slight change to maintain previous logic: 
`self.smart_proxy_sync_histories.where.not(repository_id: repos).delete_all`

**Review by ianballou — COMMENTED** [review]
Looks like tests broke, but the change on a running machine works well for me.

**Comment by sjha4** [comment]
All relevant failures..Will update tests..

**Review by ianballou — APPROVED** [review]

**Pull request merged**